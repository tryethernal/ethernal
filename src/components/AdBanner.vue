<!-- AdBanner.vue -->
<template>
  <div class="ad-banner">
    <div class="sevioads" :data-zone="zoneId"></div>
  </div>
</template>

<script setup>
import { onMounted, onBeforeUnmount } from 'vue'

// Constants
const zoneId = '267ecf65-8e8c-45db-9b54-ad03098ebdb1'
const inventoryId = '7e76f5b3-3595-4e7e-adb2-763064612e3c'
const accountId = '4c3c03c6-9c27-4ab4-83d0-fd528c08cdcc'

// Lifecycle hooks
onMounted(() => {
  // Load the script
  const script = document.createElement('script')
  script.src = 'https://cdn.adx.ws/scripts/loader.js'
  script.async = true
  document.head.appendChild(script)

  // Initialize sevioads
  window.sevioads = window.sevioads || []
  const sevioads_preferences = [{
    zone: zoneId,
    adType: 'banner',
    inventoryId,
    accountId
  }]
  window.sevioads.push(sevioads_preferences)
})

onBeforeUnmount(() => {
  // Clean up script when component is destroyed
  const scripts = document.getElementsByTagName('script')
  for (let script of scripts) {
    if (script.src === 'https://cdn.adx.ws/scripts/loader.js') {
      script.remove()
    }
  }
})
</script>

<style scoped>
.ad-banner {
  width: 100%;
  min-height: 90px; /* Adjust based on your ad size */
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
}
</style>
