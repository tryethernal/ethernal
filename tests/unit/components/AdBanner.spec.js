import AdBanner from '@/components/AdBanner.vue';

describe('AdBanner.vue', () => {
    let appendChildSpy;
    let getElementsByTagNameSpy;
    let scriptElement;
    let originalCreateElement;

    beforeEach(() => {
        // Store original createElement
        originalCreateElement = document.createElement;

        // Create a real script element for our ad script
        scriptElement = originalCreateElement.call(document, 'script');
        scriptElement.src = '';
        scriptElement.async = false;

        // Mock document.createElement
        document.createElement = vi.fn((type) => {
            if (type === 'script') {
                return scriptElement;
            }
            return originalCreateElement.call(document, type);
        });

        // Mock document.head.appendChild
        appendChildSpy = vi.spyOn(document.head, 'appendChild');

        // Mock document.getElementsByTagName
        getElementsByTagNameSpy = vi.spyOn(document, 'getElementsByTagName').mockReturnValue([
            { src: 'https://cdn.adx.ws/scripts/loader.js', remove: vi.fn() }
        ]);

        // Mock window.sevioads
        window.sevioads = [];
    });

    afterEach(() => {
        // Restore original createElement
        document.createElement = originalCreateElement;
        vi.clearAllMocks();
    });

    it('Should show the component', async () => {
        const wrapper = mount(AdBanner);

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Should load the ad script on mount', async () => {
        mount(AdBanner);

        expect(document.createElement).toHaveBeenCalledWith('script');
        expect(appendChildSpy).toHaveBeenCalledWith(scriptElement);
        expect(window.sevioads.length).toBe(1);
        expect(window.sevioads[0]).toEqual([{
            zone: '267ecf65-8e8c-45db-9b54-ad03098ebdb1',
            adType: 'banner',
            inventoryId: '7e76f5b3-3595-4e7e-adb2-763064612e3c',
            accountId: '4c3c03c6-9c27-4ab4-83d0-fd528c08cdcc'
        }]);
    });

    it('Should remove the script on unmount', async () => {
        const wrapper = mount(AdBanner);
        wrapper.unmount();

        expect(getElementsByTagNameSpy).toHaveBeenCalledWith('script');
    });
});
