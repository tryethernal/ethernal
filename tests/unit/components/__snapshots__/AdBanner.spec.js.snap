// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AdBanner.vue > Should show the component 1`] = `
"<div data-v-5ae8ec01="" class="ad-banner">
  <div data-v-5ae8ec01="" class="sevioads" data-zone="267ecf65-8e8c-45db-9b54-ad03098ebdb1"></div>
</div>"
`;
